Checkout: remove handleCheckoutCompleteRedirect

sirbrillig opened the pull request:
#### Changes proposed in this Pull Request

This removes the `handleCheckoutCompleteRedirect` function from old checkout's `Checkout` component. The last place it was being used was the (disabled) post-purchase G Suite upsell nudge and that usage has been removed with https://github.com/Automattic/wp-calypso/pull/48438.

This should help confusion around where the post-purchase thank-you page comes from, since it will now always be generated by [getThankYouPageUrl](https://github.com/Automattic/wp-calypso/blob/d8b92d23138cf839b435fa63d8d0c67e19da7e17/client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts#L38).

#### Testing instructions

None. Make sure the removed code isn't in use anywhere.

sirbrillig committed:
Remove handleCheckoutCompleteRedirect

sirbrillig committed:
Remove unused dependencies of handleCheckoutCompleteRedirect

sirbrillig committed:
Remove unused props and state from Checkout

pottedmeat commented on client/my-sites/checkout/checkout/index.jsx:
```diff
@@ -2,73 +2,26 @@
-import page from 'page';
 import PropTypes from 'prop-types';
 import React from 'react';
 // eslint-disable-next-line no-restricted-imports
```
This can be deleted as well.

pottedmeat reviewed (commented):
This removes a number of props from `<Checkout>` that are still passed by `<CheckoutContainer>` that could be removed:

- couponCode
- returnToBlockEditor
- returnToHome
- selectedFeature
- purchaseId
- plan
- reduxStore
- redirectTo
- upgradeIntent
- hideNudge
- infoMessage

It also looks like there are props used in `<Checkout>` (and passed by `<CheckoutContainer>`/`<CartData>`) but not in `propTypes`:

- cart
- product
- setHeaderText
- isFetchingStoredCards

pottedmeat commented:
I think it's also worth removing the class export.

pottedmeat commented:
Is it worth worrying about any of this stuff? It looks like we might pull out the rest of this code soon.

sirbrillig committed:
Remove unnecessary eslint comment

sirbrillig commented:
Good eye. I rebased and I think we've finally got all of the code in this directory, but it's a lot, so I've split it into two PRs. This one will just remove `handleCheckoutCompleteRedirect` since I'm certain that's not being used and https://github.com/Automattic/wp-calypso/pull/48512 will handle `Checkout` and `CheckoutContainer`.

pottedmeat commented:
Cool. There's also some unused stuff in the mapped prop functions in connect.
